src/libgap-api.h

```diff
@@ -1,24 +1,113 @@
-// LibGAP API - API for using GAP as shared library.
+//// LibGAP API - API for using GAP as shared library.
 
 #ifndef LIBGAP_API_H
 #define LIBGAP_API_H
 
-#include "gap.h"
+#include "system.h"
 
-typedef void (*CallbackFunc)(void);
 
-// Initialisation and finalization
+////
+//// Setup and initialisation
+////
 
-void GAP_Initialize(int          argc,
-                    char **      argv,
-                    char **      env,
-                    CallbackFunc markBagsCallback,
-                    CallbackFunc errorCallback);
+typedef void (*GAP_CallbackFunc)(void);
+
+extern void GAP_Initialize(int              argc,
+                           char **          argv,
+                           char **          env,
+                           GAP_CallbackFunc markBagsCallback,
+                           GAP_CallbackFunc errorCallback);
+
+
+////
+//// program evaluation and execution
+////
+
+// Evaluate a string of GAP commands
+//
+// To see an example of how to use this function see tst/testlibgap/basic.c
+extern Obj GAP_EvalString(const char * cmd);
+
+
```

Comment:
One major thing missing here is a wrapper for `CallFuncList`. Actually, two variants could be interesting (and perhaps both should be provided): one is a straight wrapper for `CallFuncList`, which hence takes a plist of arguments as input. The other is a variant `CallFuncListWithArgsArray(Obj func, UInt narg, Obj * args)` which gets a pointer to an array of args. The latter can be more convenient and efficient in practice. Depends a bit on the client, though... Wonder what SageMath people would prefer, if at all.

Reply:
That wrapper was in muliple versions of the API, but without the efficiency improvement with `Obj * args`. Arguably this is *the* most important API function because once one can create primitive values and have access to global variables one can do everything with GAP.

Reply:
Agreed. The `Obj * args*` is the most likely for use in GAPJulia's `JuliaInterface -- subject to benchmarking, we might also stay with just using the non-official API, like any other GAP kernel extension (for obvious reasons, I am personally not very concerned about using some kind of "unofficial" API...). So I would use this `CallFuncList` if the measured overhead over our current solution is negligible. But I'd be willing to provide this API anyway, but only if somebody else is interested in using it... E.g. you, or the SageMath folks, or somebody else.

Reply:
Ah man, I caved in and added `GAP_CallFuncList` and `GAP_CallFuncArray`.